runtime/smithy-client/common/src/aws/smithy/kotlin/runtime/client/SdkClientOption.kt

```diff
@@ -31,6 +31,11 @@ public object SdkClientOption {
      * The client logging mode (see [SdkLogMode]
      */
     public val LogMode: ClientOption<SdkLogMode> = ClientOption("LogMode")
+
+    /**
+     * Whether endpoint discovery is enabled or not. Default is true
+     */
+    public val EndpointDiscoveryEnabled: ClientOption<Boolean> = ClientOption("EndpointDiscoveryEnabled")
```

Comment:
**Question**: Why was this added?

Reply:
Moved from `aws-sdk-kotlin`. We can remove it right now completely since it's unused but it was solving a FIXME from `aws-sdk-kotlin`.